pkg/ebpf/c/tracee.bpf.c

```diff
@@ -6289,3 +6312,836 @@ int tc_ingress(struct __sk_buff *skb)
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 1024);          // simultaneous cgroup/skb ingress/eggress progs
+    __type(key, u64);                   // sk_buff timestamp
+    __type(value, net_event_context_t); // event context built so cgroup/skb can use
+} cgrpctxmap SEC(".maps");              // saved info between SKB caller and SKB program
+
+// inodemap
+
+typedef struct net_task_context {
+    struct task_struct *task;
+    task_context_t taskctx;
+} net_task_context_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __uint(max_entries, 10240);             // simultaneous sockets being traced
+    __type(key, u64);                       // socket inode number ...
+    __type(value, struct net_task_context); // ... linked to a task context
+} inodemap SEC(".maps");                    // relate sockets and tasks
+
+// entrymap
+
+typedef struct entry {
+    long unsigned int args[6];
+} entry_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 1024);       // simultaneous tasks being traced for entry/exit
+    __type(key, u32);                // host thread group id (tgid or tid) ...
+    __type(value, struct entry);     // ... linked to entry ctx->args
+} entrymap SEC(".maps");             // can't use args_map (indexed by existing events only)
+
+// clang-format on
+
+//
+// support functions for network code
+//
+
+static __always_inline u64 sizeof_net_event_context_t(void)
+{
+    return sizeof(net_event_context_t) - 7; // neteventctx metadata & padding
+}
+
+static __always_inline void get_net_task_context(event_data_t *data, net_task_context_t *netctx)
+{
+    netctx->task = data->task;
+    __builtin_memcpy(&netctx->taskctx, &data->context.task, sizeof(task_context_t));
+}
+
+static __always_inline void should_submit_net_event(net_event_context_t *neteventctx,
+                                                    config_entry_t *config)
+{
+    // configure network events that should be sent to userland
+
+    if (should_submit(NET_PACKET_IP, config))
+        neteventctx->should_submit |= SUB_NET_PACKET_IP;
+
+    if (should_submit(NET_PACKET_TCP, config))
+        neteventctx->should_submit |= SUB_NET_PACKET_TCP;
+
+    if (should_submit(NET_PACKET_UDP, config))
+        neteventctx->should_submit |= SUB_NET_PACKET_UDP;
+
+    if (should_submit(NET_PACKET_ICMP, config))
+        neteventctx->should_submit |= SUB_NET_PACKET_ICMP;
+
+    if (should_submit(NET_PACKET_ICMPV6, config))
+        neteventctx->should_submit |= SUB_NET_PACKET_ICMPV6;
+
+    if (should_submit(NET_PACKET_DNS, config))
+        neteventctx->should_submit |= SUB_NET_PACKET_DNS;
+}
+
+//
+// Socket Creation: keep track of created socket inodes per traced task
+//
+
+SEC("kprobe/sock_alloc_file")
+int BPF_KPROBE(trace_sock_alloc_file)
+{
+    // runs every time a socket is created (entry)
+
+    event_data_t data = {};
+
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    if (!should_trace(&data))
```

Comment:
Yep, I know what you mean... We should have a "lighter" version of the init_event_data for network hooks (probably). I can add that as a task of my EPIC if you are okay.

Note: I also have to check situations when tasks are moved from one cgroup to another. The socket <=> task relation would be set to the task with old context (not the one re-calculated). I think I would pick the relation from egress (like if it was an existing socket), but needs to confirm, will add that as a task as well.

Reply:
I believe the comment was mostly because we are not taking in consideration (yet ?) the change of scopes for the new networking code (part of another discussion, in one of your comments). Either way, I'm using the interface we have available for all other ebpf programs, and I'll keep it like that for this PR, so I'm marking this as "solved" and we can discuss this later if ever needed.